rust-client/src/bin/delegated_prover.rs

```diff
@@ -0,0 +1,106 @@
+use std::sync::Arc;
+
+use miden_client::crypto::FeltRng;
+use miden_client::{
+    asset::FungibleAsset,
+    keystore::FilesystemKeyStore,
+    note::NoteType,
+    rpc::Endpoint,
+    transaction::{OutputNote, TransactionProver, TransactionRequestBuilder},
+    ClientError, Felt, RemoteTransactionProver,
+};
+use miden_client_tools::{
+    create_basic_account, create_basic_faucet, create_exact_p2id_note, instantiate_client,
+    mint_from_faucet_for_account,
+};
+
+#[tokio::main]
+async fn main() -> Result<(), ClientError> {
+    // Initialize client, keystore, & delegated prover endpoint
+    let endpoint = Endpoint::testnet();
+    let mut client = instantiate_client(endpoint, None).await.unwrap();
+
+    let keystore: FilesystemKeyStore<rand::prelude::StdRng> =
+        FilesystemKeyStore::new("./keystore".into()).unwrap();
+
+    let remote_tx_prover: RemoteTransactionProver =
+        RemoteTransactionProver::new("https://tx-prover.testnet.miden.io");
+    let tx_prover: Arc<dyn TransactionProver + 'static> = Arc::new(remote_tx_prover);
+
+    let sync_summary = client.sync_state().await.unwrap();
+    println!("Latest block: {}", sync_summary.block_num);
+
+    let (alice_account, _) = create_basic_account(&mut client, keystore.clone())
+        .await
+        .unwrap();
+
+    let (bob_account, _) = create_basic_account(&mut client, keystore.clone())
+        .await
+        .unwrap();
+
+    let faucet = create_basic_faucet(&mut client, keystore).await.unwrap();
+
+    let _ = mint_from_faucet_for_account(&mut client, &alice_account, &faucet, 1000, None)
+        .await
+        .unwrap();
+
+    let account = client
+        .get_account(alice_account.id())
+        .await
+        .unwrap()
+        .unwrap();
+
+    println!(
+        "Alice Account balance: {:?}",
+        account.account().vault().get_balance(faucet.id())
+    );
+
+    // Creating 10 P2ID notes with 10 tokens each to send to Bob
```

Comment:
So here you are sending 10 P2ID notes in one transaction, right? Can you make this more explicit?